packages/authorization-platformstate-writer/src/consumerServiceV2.ts

```diff
@@ -300,43 +253,38 @@ export async function handleMessageV2(
           );
 
         const addedTokenGenStatesConsumerClients: TokenGenerationStatesConsumerClient[] =
-          [];
-
-        await Promise.all(
-          client.keys.map(async (key) => {
-            const newTokenGenStatesConsumerClient: TokenGenerationStatesConsumerClient =
-              createTokenGenStatesConsumerClient({
-                consumerId: client.consumerId,
-                kid: key.kid,
-                publicKey: key.encodedPem,
-                clientId: client.id,
-                purposeId,
-                purposeEntry,
-                agreementEntry,
-                catalogEntry,
-              });
+          await Promise.all(
+            client.keys.map(async (key) => {
+              const newTokenGenStatesConsumerClient: TokenGenerationStatesConsumerClient =
```

Comment:
down [here](https://github.com/pagopa/interop-be-monorepo/pull/1553/files?diff=split&w=1#diff-9543841486f8c58ff2afaac58a4de67f687bb4ee33fce443421156feae0bf926R260) should be the consumer of the purpose